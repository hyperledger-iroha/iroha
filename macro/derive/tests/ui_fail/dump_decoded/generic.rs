use parity_scale_codec::Decode;

#[derive(Debug, Decode, iroha_derive::DumpDecoded)]
struct Struct<T>;
