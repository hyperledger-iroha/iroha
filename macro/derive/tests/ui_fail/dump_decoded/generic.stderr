error: proc-macro derive panicked
 --> tests/ui_fail/dump_decoded/generic.rs:3:25
  |
3 | #[derive(Debug, Decode, iroha_derive::DumpDecoded)]
  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Implementation of `DumpDecoded` for generic has no meaning

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui_fail/dump_decoded/generic.rs:1:1
  |
1 | / use parity_scale_codec::Decode;
2 | |
3 | | #[derive(Debug, Decode, iroha_derive::DumpDecoded)]
4 | | struct Struct<T>;
  | |_________________^ consider adding a `main` function to `$DIR/tests/ui_fail/dump_decoded/generic.rs`

error[E0392]: parameter `T` is never used
 --> tests/ui_fail/dump_decoded/generic.rs:4:15
  |
4 | struct Struct<T>;
  |               ^ unused parameter
  |
  = help: consider removing `T`, referring to it in a field, or using a marker such as `PhantomData`
  = help: if you intended `T` to be a const parameter, use `const T: usize` instead
